Sources/LeafKit/LeafLexer.swift

```diff
@@ -1,147 +1,210 @@
-struct LeafLexer {
-    private var buffer: ByteBuffer
+extension ByteBuffer {
+    mutating func stringify() -> String? {
+        return readString(length: readableBytes)
+    }
+}
+
+extension UInt8 {
+    var isValidInTagName: Bool {
+        return isLowercaseLetter
```

Comment:
For Vapor 4, I've been adopting some things from NIO's code style like always using `self.`. It makes the code a lot easier to reason about in places without context sensitive syntax-highlighting (like GitHub)